src/test/versionbits_tests.cpp

```diff
@@ -447,6 +444,15 @@ BOOST_AUTO_TEST_CASE(versionbits_computeblockversion)
         }
     }
 
+    {
+        // Use regtest/testdummy to ensure we always exercise some
+        // deployment that's not always/never active
+        ArgsManager args;
+        args.ForceSetArg("-vbparams", "testdummy:1199145601:1230767999"); // January 1, 2008 - December 31, 2008
```

Comment:
55ac5f568a3b73d6f1ef4654617fb76e8bcbccdf: why not just use `-2` here?

Reply:
-2 would be never active, this is to ensure we execute the test code even if all the actual deployments are always active or never active, to avoid the test code bit rotting

Reply:
Maybe add an explicit test for -1/-2 as well?

Reply:
Currently taproot on regtest/signet tests the always active (-1) case, and testdummy on mainnet should always test the never active (-2) case, so think this can be left for later cleanup/refactoring.

Reply:
Sure, no need to force push. I am just dumping ideas.

Reply:
@ajtowns maybe I'm missing something, but `January 1, 2008 - December 31, 2008` is also never active. Or do we use earlier fake dates in the test?

Reply:
@Sjors Note that with the updated state machine, it's possible for a real chain (not just a fake one in the unit tests) to become ACTIVE even if starttime/timeout are prior to the genesis block's timestamp: in that case, the first period will be DEFINED, the second STARTED, and the third either LOCKED_IN or FAILED depending on signalling in the second period. Not an issue for chains that are already well past genesis though, so real is limited to regtest, brand new signets, or someone trying to reorg mainnet/testnet3 all the way to genesis.